kernel/runtime/crt-init/runtime/crtbegin.cc

```diff
@@ -24,14 +24,19 @@
  * Copyright 2016 Randolf Rotta, Robert Kuban, and contributors, BTU Cottbus-Senftenberg
  */
 
-#include "stdlib.h"
+#include <cstdlib>
 #include <cstddef>
 #include <atomic>
 
 #include "mythos/syscall.hh"
 #include "runtime/mlog.hh"
 #include "runtime/tls.hh"
 
+extern "C" typedef void (*atexit_func_t)();
+extern "C" int atexit(atexit_func_t func);
```

Comment:
gcc version 5.4.0 20160609 requires either to remove this declaration or to add `noexcept` to the definition (line 84) in order to compile.